enterprise/views.py

```diff
@@ -933,3 +949,272 @@ def get(self, request, enterprise_uuid, course_id):
+        ).format(count=course_count)
+        effort_info_text = ungettext_min_max(
+            '{} hour per week, per course',
+            '{} hours per week, per course',
+            self.context_data['effort_info_text'],
+            program_details.get('min_hours_effort_per_week'),
+            program_details.get('max_hours_effort_per_week'),
+        )
+        length_info_text = ungettext_min_max(
+            '{} week per course',
+            '{} weeks per course',
+            self.context_data['length_info_text'],
+            program_details.get('weeks_to_complete_min'),
+            program_details.get('weeks_to_complete_max'),
+        )
+
+        # Update some enrollment-related text requirements.
+        if program_details['enrolled_in_program']:
+            purchase_action = self.actions['purchase_unenrolled_courses']
+            item = self.items['enrollment']
+        else:
+            purchase_action = self.actions['purchase_program']
+            item = self.items['program_enrollment']
+
+        # Add any warning messages.
+        program_data_sharing_consent = get_data_sharing_consent(
+            request.user.username,
+            enterprise_customer.uuid,
+            program_uuid=program_details['uuid'],
+        )
+        if program_data_sharing_consent.exists and not program_data_sharing_consent.granted:
+            add_consent_declined_message(request, enterprise_customer, program_title)
+
+        discount_data = program_details.get('discount_data', {})
+        if discount_data.get('total_incl_tax_excl_discounts') is None:
+            add_missing_price_information_message(request, program_title)
+
+        one_click_purchase_eligibility = program_details.get('is_learner_eligible_for_one_click_purchase', False)
+        if not one_click_purchase_eligibility:
+            add_not_one_click_purchasable_message(request, enterprise_customer, program_title)
+
+        # Update our context with the above calculated details and more.
+        context_data = self.context_data.copy()
+        context_data.update(get_global_context(request))
+        context_data.update({
+            'enterprise_welcome_text': self.context_data['enterprise_welcome_text'].format(
+                strong_start='<strong>',
+                strong_end='</strong>',
+                enterprise_customer_name=enterprise_customer.name,
+                platform_name=platform_name,
+            ),
+            'discount_provider': self.context_data['discount_provider'].format(
+                strong_start='<strong>',
+                strong_end='</strong>',
+                provider=enterprise_customer.name,
+            ),
+            'enterprise_customer': enterprise_customer,
+            'organization_name': organization.get('name'),
+            'organization_logo': organization.get('logo_image_url'),
+            'organization_text': self.context_data['organization_text'].format(organization=organization['name']),
```

Comment:
Crash with missing org isn't quite fixed; the `organization` doesn't have a `name` key if it's missing above. This is a relatively minor thing, but I want to make sure that we're building something that's not only suitable for edX.

Reply:
Ah, thanks :+1: